morpheus/_lib/include/morpheus/stages/preprocess_fil.hpp

```diff
@@ -76,23 +88,234 @@ class PreprocessFILStage
     std::string m_vocab_file;
 };
 
+template <typename InputT, typename OutputT>
+PreprocessFILStage<InputT, OutputT>::PreprocessFILStage(const std::vector<std::string>& features) :
+  base_t(base_t::op_factory_from_sub_fn(build_operator())),
+  m_fea_cols(std::move(features))
+{}
+
+template <typename InputT, typename OutputT>
+PreprocessFILStage<InputT, OutputT>::subscribe_fn_t PreprocessFILStage<InputT, OutputT>::build_operator()
+{
+    return [this](rxcpp::observable<sink_type_t> input, rxcpp::subscriber<source_type_t> output) {
+        return input.subscribe(rxcpp::make_observer<sink_type_t>(
+            [&output, this](sink_type_t x) {
+                auto next = this->pre_process_batch(x);
+                output.on_next(std::move(next));
+            },
+            [&](std::exception_ptr error_ptr) {
+                output.on_error(error_ptr);
+            },
+            [&]() {
+                output.on_completed();
+            }));
+    };
+}  // namespace morpheus
+
+void transform_bad_columns(std::vector<std::string>& fea_cols,
+                           morpheus::MutableTableInfo& mutable_info,
+                           std::vector<std::string>& df_meta_col_names);
```

Comment:
What is this function for?

- If its a public global namespaced function, it should be in one of the utils files.
- If its tied to this particular class, make it a static function.
- And if it is a public function, it needs to have doxygen comments.

Reply:
I abstracted this out from the original `fix_bad_columns()` to avoid duplicate code. Do you have any specific reason that this should be a static function rather than non-static?